core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -226,6 +228,74 @@ public void cancelOperation() throws UnsupportedOperationException {
         return this.fluxHandle.blockLast();
     }
 
+    /**
+     * Blocks until given {@link OperationStatus} is received or a timeout expires.
+     * @param statusToBlockFor The {@link OperationStatus} to block for and it can be any valid {@link OperationStatus} except {@link OperationStatus#OTHER}.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtil(PollResponse.OperationStatus statusToBlockFor, Duration timeout) {
```

Comment:
I would suggest having an overload for both blockUntil methods that doesn't take a Duration which will block indefinitely.